### PR TITLE
GT-1501 Add a couple new analytics events for IAM targetting

### PR DESCRIPTION
**Pull request by frett**
- trigger the all tools analytics event from the new all tools UI
- remove the unused MODE_ALL for ToolsListFragment
- add a couple new analytics events to be targeted by iam

### CODE REVIEW

File: app/src/main/kotlin/org/cru/godtools/ui/tools/ToolsListFragment.kt [hunk 1]
```diff
@@ -116,12 +115,14 @@ class ToolsListFragment() : BasePlatformFragment<ToolsFragmentBinding>(R.layout.
 
     private fun trackInAnalytics() {
         when (mode) {
-            MODE_ALL -> eventBus.post(AnalyticsScreenEvent(SCREEN_ALL_TOOLS))
```
Comment (gyasistory) [inline app/src/main/kotlin/org/cru/godtools/ui/tools/ToolsListFragment.kt:119]: @frett is MODE_ALL no longer being used.

File: app/src/main/kotlin/org/cru/godtools/ui/tools/ToolsListFragmentDataModel.kt [hunk 2]
```diff
@@ -49,7 +48,6 @@ class ToolsListFragmentDataModel @Inject constructor(private val dao: GodToolsDa
         when {
             mode == MODE_ADDED && !training && PageSet.TRAINING.supportsLocale(Locale.getDefault()) ->
                 BannerType.TUTORIAL_TRAINING
-            mode == MODE_ALL && !favorite -> BannerType.TOOL_LIST_FAVORITES
```
Comment (gyasistory) [inline app/src/main/kotlin/org/cru/godtools/ui/tools/ToolsListFragmentDataModel.kt:52]: Same question

**Review by gyasistory — CHANGES_REQUESTED** [review]
Just a minor question

File: app/src/main/kotlin/org/cru/godtools/ui/tools/ToolsListFragment.kt (hunk 1 above)
Comment (frett) [reply app/src/main/kotlin/org/cru/godtools/ui/tools/ToolsListFragment.kt:119]: https://github.com/CruGlobal/godtools-android/pull/2246/commits/00a6048cd71a3a744bb1c213a47cd73f7af32a11

yes, it was no longer needed after the all tools UI was switched to the new ToolsFragment.

**Review by gyasistory — APPROVED** [review]

**Pull request merged**